### ISSUE TITLE
Unable to read core_company table

### ISSUE THREAD

**Issue by rohantjoshi** [gflewis/sndml#2]
Hello, we are in very weird situation. We are following all the steps mentioned and we are able to get the data of core_company for one of the sub_prod instance(QA). 
Same when we are trying for another sub_prod instance(stage) its is working for all other table except core_company.
when we enable soap logs below is what we are seeing : 

<?xml version="1.0" encoding="UTF-8"?><env:Envelope xmlns:env="http://schemas.xmlsoap.org/soap/envelope/" xmlns:enc="http://schemas.xmlsoap.org/soap/encoding/" xmlns:tns="http://www.service-now.com/core_company"><env:Header/><env:Body><tns:getRecords><__limit>1</__limit></tns:getRecords></env:Body></env:Envelope>

Any thoughts or comments we should do to debug this.
P.S. : We have compared ACL's on both instances both are same, even tried logging to ServiceNow with the user we are trying to authenticate the DataMart and we are able to see core_company with in servicenow.
But it is not getting extracted.

**Comment by gflewis** [comment]
Before starting the load the code is reading a single record to verify that the table is readable. You can shut this off by adding the following line to the properties file:
`datamart.check_readable=false`

This will not solve your problem with core_company, but it should eliminate one source of noise.

The peculiar thing is that the "checkReadable" test is not aborting the program. This code throws an exception if it cannot read at least one record from the table. I wonder if you are getting a SOAP timeout. If this is the case then the program will issue a "getKeys" request and then appear to hang. How many records are there in the core_company table?

I have created a blog post which may assist you in diagnosing the issue.
https://servicenowsoap.wordpress.com/2017/06/14/sndml-debugging/

Please let me know what you learn.

**Comment by rohantjoshi** [comment]
We tried adding the property and it worked for us. Not sure exactly what is happening but it is working.
Can you please explain what that property does internally ?

**Comment by gflewis** [comment]
I have not been able to reproduce your error. I have updated the wiki with a description of the check_readable property. I am closing this issue. Please open a new issue if you can provide me with additional information which might enable me to reproduce a potential problem with the checkReadable test. Thanks.
